lib/rock/packaging/packageinfoask.rb

```diff
@@ -0,0 +1,177 @@
+
+module Autoproj
+    module Packaging
+        class PackageInfoAsk
+
+            class << self
+                alias :class_new :new
+
+                def new(which, options)
```

Comment:
Ok, I should look closer ;). Why did you opt for this solution vs. using initialize with an optional parameter for the selection of the particular adaptor implementation, and delegating the query calls?